src/BookingForm.js

```diff
@@ -0,0 +1,153 @@
+import { faCheckCircle } from "@fortawesome/free-solid-svg-icons";
+import "./BookingForm.css";
+
+const BookingForm = props => {
+  const [id, setId] = useState("");
+  const [title, setTitle] = useState("");
+  const [firstName, setFirstName] = useState("");
+  const [surname, setSurname] = useState("");
+  const [email, setEmail] = useState("");
+  const [checkInDate, setCheckInDate] = useState("");
+  const [checkOutDate, setCheckOutDate] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    let error = "";
+
+    if (firstName.trim() === "") {
+      error = "Please enter your first name .";
+      setErrorMessage(error);
+    } else if (surname.trim() === "") {
+      error = "Please enter your surname .";
+      setErrorMessage(error);
+    } else if (
+      email.indexOf("@") < 1 ||
+      email.lastIndexOf(".") < email.indexOf("@") + 2 ||
+      email.lastIndexOf(".") + 2 >= email.length
+    ) {
+      error = "Please enter a valid email address.";
+      setErrorMessage(error);
+    } //else if (!checkInDate) {
+    //const today = new Date();
+    // setCheckInDate(today.toLocaleDateString("en-GB"));
+    //}
+    else {
+      let newRow = {
+        id: id,
+        title: title,
+        firstName: firstName,
+        surname: surname,
+        email: email,
+        checkInDate: checkInDate,
+        checkOutDate: checkOutDate
+      };
+      //setNewBookings([...newBookings, newRow]);
+      props.onAddBooking(newRow);
+
+      setTitle("");
+      setFirstName("");
+      setSurname("");
+      setEmail("");
+      setId("");
+      setCheckInDate("");
+      setCheckOutDate("");
+    }
+  }
+  // const search = (searchVal) => {
+  //   console.info("TO DO!", searchVal);
+  //   setNewBookings(
+  //     newBookings.filter(
+  //       (booking) =>
+  //         booking.firstName.toLowerCase().includes(searchVal.toLowerCase()) ||
+  //         booking.surname.toLowerCase().includes(searchVal.toLowerCase())
+  //     )
+  //   );
+  // };
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit}>
+        <input
+          type="number"
+          name="id"
+          placeholder="id"
+          value={id}
+          onChange={event => setId(event.target.value)}
+        />
+        <input
+          type="text"
+          name="title"
+          placeholder="Title"
+          value={title}
+          onChange={event => setTitle(event.target.value)}
+        />
+        <div className="firstName">
+          <input
+            type="text"
+            name="firstName"
+            placeholder="FirstName"
+            value={firstName}
+            onChange={event => setFirstName(event.target.value)}
+          />
+          {!firstName.trim() && <span>X first name must not be empty</span>}
+          {firstName.trim() && (
+            <FontAwesomeIcon icon={faCheckCircle} style={{ color: "green" }} />
+          )}
```

Comment:
Let's use a ternary operator instead  - look at doing this in other places too
```suggestion
          {firstName.trim() ? (
            <FontAwesomeIcon icon={faCheckCircle} style={{ color: "green" }} />
          ) : <span>X first name must not be empty</span>}
```